scripts/dts/edtlib.py

```diff
@@ -773,7 +773,17 @@ def _init_binding(self):
                 # specified bus (if any) and then against any bus. This is so
                 # that matching against bindings which do not specify a bus
                 # works the same way in Zephyr as it does elsewhere.
-                if (compat, on_bus) in self.edt._compat2binding:
+                if isinstance(on_bus, list) \
+                    and ((compat, elem) in self.edt._compat2binding for elem in on_bus):
```

Comment:
Does this work?

It looks like the predicate is `isinstance(...) and (generator_expression)`. Since a generator expression is always truthy, this test doesn't look right.

Did you mean to use `any((compat, elem) in ...)` or something like that?

Would be good to get some test cases where there is and is not a matching compat,elem pair in self.edt._compat2binding just to make sure and avoid regressions.

Reply:
yes, you are correct. I will update it

Reply:
In order for me to add test cases, I would have to add extra *-bus.yaml files. It doesn't look like this was being tested before 😕

Reply:
>  I would have to add extra *-bus.yaml files

The current test cases were bulk ported from a pre-pytest suite that was hand-rolled.

If you want to create your own bindings directories and do special-purpose tests with DTS files that are specific to just testing this new feature to make your life easier, that would be fine with me.

Reply:
@mbolivar A test case was added to test_bus(), testing the priority in the list